detect_meteors.py

```diff
@@ -176,11 +182,52 @@ def detect_meteors(rf_dir, id_dir, noise_dir, output_dir,
         tmm.loadFromHdf5('/tmp/tmm.hdf5', skip_lowlevel=True)
     else:
         tmm.loadFromHdf5(skip_lowlevel=True)
-
-    for k, (tx, rx) in enumerate(data_generator(rfo, ido, no, tmm, s0, s1, rxch, txch)):
-        #FIXME call processing functions here
-        pass
-
+    
+    data_list = []
+    saved_data = np.zeros((59858, 1120))
+    pulse_data = data_generator(rfo, ido, no, tmm, s0, s1, rxch, txch)
+    times = []
+    for k, (tx, rx) in enumerate(pulse_data):
+
+        num_of_pulses = np.int((s1 - s0)/(rx.sample_rate*0.0267300605774))
+        # function that calculates snr values
+        def data_cal(array):
+            snr_vals = (np.abs(array.values)**2)/rx.noise_power
+            snr_point = np.unravel_index(np.argmax(snr_vals), (6879, 480))
+            max_snr = snr_vals[snr_point[0], snr_point[1]]
+            return max_snr, snr_point, snr_vals
+
+        mf_rx = freq_dft(tx, rx)
+        max_snr, snr_point, snr_vals = data_cal(mf_rx)
+        meteor_list = is_there_a_meteor(mf_rx, max_snr, snr_point, 1, 20533, 211200)
+
+        def list_filter(data):
+	    velocity = mf_rx.frequency.values[snr_point[1]]*3e8/(440e6*2)
+            info = (velocity, max_snr, k)
+            data.extend(info)
+            return data
+
+        if meteor_list != []:
+            new_list = list_filter(meteor_list)
+            data_list.append(new_list) 
```

Comment:
Instead of appending this extra data to the meteor list, add it in the is_there_a_meteor function. You'll need to add the radar center frequency as an argument, and that will also let you pass speed thresholds instead of frequency and do the conversion inside the function.